lib/jekyll/site.rb

```diff
@@ -253,7 +253,7 @@ def categories
     #
     # Returns the Hash to be hooked to site.data.
     def site_data
-      config["data"] || data
+      @site_data ||= config["data"] || data
```

Comment:
If you want to memoize the `||`, then it'd be: `@site_data ||= (config["data"] || data)`. By default, this is equivalent to `(@site_data ||= config["data"]) || data`.

Reply:
This `@site_data` should also be reset in `reset`!

Reply:
Regarding the use of parenthesis, I did not know, this would be interpreted that way.. Thanks!

Reply:
@ashmaroli You might enjoy http://tmm1.net/ruby21-profiling/ which has an example of where parentheses went awry!

Reply:
Thanks. I'll check it out :smiley: